lib/web/fetch/index.js

```diff
@@ -120,8 +122,51 @@ class Fetch extends EE {
   }
 }
 
+// publishIfContext will check if there is context if so,
+// it will add error in the case of an error before publishing
+// the event if not it will return right away
+function publishIfContext (name, context, error) {
+  if (!context) return
+  if (error) context.error = error
+  channels[name].publish(context)
+}
+
+// handlePromise will handle both fulfilled/rejected promises
+// from p and will publish events accordingly
+function handlePromise (p, context) {
+  return p.promise.then(
+    (success) => {
+      channels.asyncStart.publish(context)
+      channels.asyncEnd.publish(context)
+      return success
+    },
+    (failure) => {
+      context.error = failure
+      channels.asyncStart.publish(context)
+      channels.asyncEnd.publish(context)
+      return Promise.reject(failure)
+    })
```

Comment:
I would do the error channel publish here so you don't need to add it in a bunch of place that occur right before calling `handlePromise` anyway.

```suggestion
    (failure) => {
      context.error = failure
      channels.fetchError.publish(context)
      channels.asyncStart.publish(context)
      channels.asyncEnd.publish(context)
      return Promise.reject(failure)
    })
```